### ISSUE TITLE
minetest.after is unreliable

### ISSUE THREAD

**Issue by sofar** [minetest/minetest#3592]
Minetest.after is a trivial function that promises to schedule a function with a resolution of ~1 second. However, in my tests it shows that on a completely idle server, no mods, it fails to schedule even within 2-3 seconds of the requested interval.

How I measured:

```
diff --git a/mods/default/init.lua b/mods/default/init.lua
index 6f1b148..2f86513 100644
--- a/mods/default/init.lua
+++ b/mods/default/init.lua
@@ -46,3 +46,16 @@ dofile(minetest.get_modpath("default").."/player.lua")
 dofile(minetest.get_modpath("default").."/trees.lua")
 dofile(minetest.get_modpath("default").."/aliases.lua")
 dofile(minetest.get_modpath("default").."/legacy.lua")
+
+local lasttime = os.time()
+local function timertest()
+       local newtime = os.time()
+       local diff = newtime - lasttime
+       print("elapsed: " .. diff)
+       lasttime = os.time()
+       local new = math.random(1,5)
+       print("next: " .. new)
+       minetest.after(new, timertest)
+end
+
+minetest.after(1, timertest)

```

The measured data:
```
next: 5 elapsed: 6
next: 4 elapsed: 6
next: 2 elapsed: 4
next: 5 elapsed: 7
next: 3 elapsed: 5
next: 5 elapsed: 8
next: 3 elapsed: 6
next: 1 elapsed: 5
next: 1 elapsed: 4
next: 3 elapsed: 6
next: 4 elapsed: 7
next: 5 elapsed: 9
next: 5 elapsed: 8
next: 4 elapsed: 7
next: 2 elapsed: 5
next: 3 elapsed: 6
next: 4 elapsed: 8
next: 5 elapsed: 8
```

In not a single sample did the timer execute within 1 seconds of the requested timeout. In most cases, it was delayed by an additional 3 seconds, sometimes even 4 seconds.

As a timer function, I expect ```minetest.after()``` to correct for time lost in lag, and immediately expire all timers that are past due, and execute them. This is not happening.

**Comment by sofar** [comment]
Measurements after I rewrote the timer code to properly account for timer lag:

```
next: 2 elapsed: 2
next: 5 elapsed: 5
next: 3 elapsed: 3
next: 3 elapsed: 3
next: 2 elapsed: 2
next: 1 elapsed: 1
next: 3 elapsed: 3
next: 4 elapsed: 4
next: 3 elapsed: 3
next: 5 elapsed: 5
next: 4 elapsed: 4
next: 4 elapsed: 4
next: 4 elapsed: 4
next: 2 elapsed: 2
next: 2 elapsed: 2
next: 2 elapsed: 2
next: 4 elapsed: 4
next: 2 elapsed: 2
next: 3 elapsed: 3
next: 4 elapsed: 4
next: 3 elapsed: 3
next: 1 elapsed: 1
next: 2 elapsed: 2
next: 1 elapsed: 1
next: 3 elapsed: 3
next: 5 elapsed: 5
next: 2 elapsed: 2
next: 5 elapsed: 5
next: 2 elapsed: 2
next: 2 elapsed: 2
next: 5 elapsed: 5
next: 4 elapsed: 4
next: 3 elapsed: 3
next: 5 elapsed: 5
```

**Comment by sofar** [comment]
fixes #3593
